tests/unit/models/DefaultPluginData.spec.js

```diff
@@ -0,0 +1,330 @@
+
+        expect(pluginData.getComponentById('test2')).toEqual(new Component({
+          id: 'test2',
+          name: 'test2',
+          definition,
+        }));
+      });
+
+      it('Should return the sub-component corresponding to the given id', () => {
+        const definition = new ComponentDefinition();
+        const pluginData = new DefaultPluginData();
+
+        pluginData.addComponent('test', definition);
+        pluginData.addComponent('test2', definition);
+        pluginData.components[0].children = [pluginData.components[1]];
+        pluginData.components = [pluginData.components[0]];
+
+        expect(pluginData.getComponentById('test2')).toEqual(new Component({
+          id: 'test2',
+          name: 'test2',
+          definition,
+        }));
+      });
+    });
+
+    describe('Test method: removeComponentById', () => {
+      it('Should remove selected component', () => {
+        const definition = new ComponentDefinition();
+        const pluginData = new DefaultPluginData();
+
+        pluginData.addComponent('test', definition);
+        pluginData.addComponent('test2', definition);
+
+        pluginData.removeComponentById('test');
+        expect(pluginData.components).toEqual([
+          new Component({
+            id: 'test2',
+            name: 'test2',
+            definition,
+          }),
+        ]);
+
+        pluginData.removeComponentById('test2');
+        expect(pluginData.components).toEqual([]);
+      });
+
+      it('Should do nothing on unknown id', () => {
+        const definition = new ComponentDefinition();
+        const pluginData = new DefaultPluginData();
+
+        pluginData.addComponent('test', definition);
+        pluginData.addComponent('test2', definition);
+
+        pluginData.removeComponentById('unknown');
+        expect(pluginData.components).toEqual([
+          new Component({
+            id: 'test',
+            name: 'test',
+            definition,
+          }),
+          new Component({
+            id: 'test2',
+            name: 'test2',
+            definition,
+          }),
+        ]);
+      });
+
+      it('Should remove existing source link', () => {
+        const pluginData = new DefaultPluginData();
+        pluginData.links.push(new ComponentLink({ source: 'root' }));
+        pluginData.links.push(new ComponentLink({ source: 'test' }));
+
+        pluginData.removeComponentById('root');
+        expect(pluginData.links).toEqual([new ComponentLink({ source: 'test' })]);
+      });
+
+      it('Should remove existing target link', () => {
+        const pluginData = new DefaultPluginData();
+        pluginData.links.push(new ComponentLink({ target: 'root' }));
+        pluginData.links.push(new ComponentLink({ target: 'test' }));
+
+        pluginData.removeComponentById('root');
+        expect(pluginData.links).toEqual([new ComponentLink({ target: 'test' })]);
+      });
+
+      it('Should remove selected depth component', () => {
```

Comment:
'Should remove the sub-component corresponding to the given id'

Reply:
done